packages/sdk/src/types/CommonTypes.ts

```diff
@@ -411,15 +419,59 @@ export interface ActionEventErrorData {
 }
 
 interface AsyncErrorBase {
+    type: AsyncErrorType;
     message: string;
 }
 
-export interface ActionAsyncError extends AsyncErrorBase {
+export class ActionAsyncError implements AsyncErrorBase {
+    type: AsyncErrorType;
     id?: string;
     event?: ActionEditorEvent;
     eventChain?: ActionEventErrorData[];
+    message: string;
+
+    constructor(message: string, id?: string, event?: ActionEditorEvent, eventChain?: ActionEventErrorData[]) {
+        this.type = AsyncErrorType.action;
+        this.message = message;
+        this.id = id;
+        this.event = event;
+        this.eventChain = eventChain;
+    }
+}
+
+export class DataRowAsyncError implements AsyncErrorBase {
+    type: AsyncErrorType;
+    count: number;
+    exceptions: EditorExceptionDto[];
+    message: string;
+
+    constructor(count: number, message: string, exceptions: EditorExceptionDto[]) {
+        this.type = AsyncErrorType.dataRow;
+        this.count = count;
+        this.message = message;
+        this.exceptions = exceptions;
+    }
+}
+
+export enum AsyncErrorType {
+    action = 'action',
+    dataRow = 'dataRow',
+}
+
+export interface ExceptionContext {
```

Comment:
Consider naming it `VariableDataRowExceptionContext`, maybe in the future we will have more types of `DataRowExceptionContext`?

```suggestion
export interface DataRowExceptionContext {
```